### ISSUE TITLE
Multi-dimensional input in LSTM

### ISSUE THREAD

**Issue by herleeyandi** [fchollet/keras#5798]
Hello I am new in using keras. Today I works on video that exploit the LSTM method in keras.

1) I have input of coordinate array let say 68 points, so it will be (68X2) it is the dimension which input per frame. Then first run I will wait until 10 frames (should it be statefull LSTM?), then output one hidden unit (binary classification). So how can I code it in keras?, For example, let assume here I have 14 frames data, so I tried like this:
`model = Sequential()`
`model.add(LSTM(1, batch_input_shape=(14, 10, 68), stateful=True, return_sequences=True))`
`model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])`
` model.summary()`  
It still give me error. How I input the data dimension of (68X2)?, should I flatten it first to (136X1)?, how about if the feature is in 3-Dimensional for example (68X2X4)?
As long as input_shape should be in format [number_of_train, sequences, dimension]

2) How we visualize the architecture of  LSTM?, I mean in here like visualize the architecture of network like in here : http://ethereon.github.io/netscope/quickstart.html which visualize the network in caffe prototxt. This will help me to make sure that my network is actually right implementation of my minds.

Thank you!

**Comment by emicol** [comment]
For the point 2 : 
```
    from keras.utils import plot_model
    plot_model(autoencoder, to_file='data/model.png')
    plot_model(autoencoder, to_file='data/model_shapes_.png', show_shapes=True)
```
For the first point, yes, the input shape is not fine. It should be simple to not use batch input (yes, with batch input set, specify stateful=True).  Flat the dimension seems best for me, but with a differentiation between x and y values (ie x between[0;99];y[100;99])
```
model = Sequential()
model.add(Input(shape=(136,)))
model.add(Embedding(nb_values_x_y, output_dim=1, input_length=136))
model.add(LSTM(1))

```
But it will be better to add neurons to your lstm, and to redirect its outputs to a dense layer with a softmax. It will allow you to get an array of probabilities, and find the best one.

Good work and sorry for my english,
Emilie
